lib/_http_parser.js

```diff
@@ -0,0 +1,890 @@
+/*
```

Comment:
two things

1. this is deprecating `process.binding('http_parser')` which means we're gonna break some people, maybe too many to be able to remove this right away.
2. if we don't expose this parser for use externally then I think it should be it's own library and vendored in like `readable-stream` is so that people can require it on npm. if we do *that* then I think it's reasonable to make it internal.